docs/SQL_SERVER_WC.md

```diff
@@ -0,0 +1,72 @@
+
+### SQL Server Connection URI
+
+A Sno repository with a SQL Server working copy needs to be configured with a `mssql://` connection URI. This URI contains how to connect to the server, the name of the database to connect to (which can be shared with other tasks), and the name of the schema that should be managed as a working copy by this Sno repository.
+
+A connection URL would generally have the following format:
+
+`mssql://[user[:password]@][netloc][:port][/dbname]`
+
+Since Sno also requires the schema to be specified up front, Sno needs a connection URL in the following format:
+
+`mssql://[user[:password]@][netloc][:port]/dbname/dbschema`
+
+For example, a Sno repo called `airport` might have a URL like the following:
+
+`mssql://sno_user:password@localhost:1433/gis/airport_sno`
+
+To configure a Sno repository to use a particular SQL Server schema as its working copy, specify the `--workingcopy` flag when creating the repository, for example:
+
+`sno init --workingcopy=mssql://... --import=...`
+
+The schema that Sno is given to manage should be either non-existent or empty at the time Sno is configured, but the server and database should already exist.
+
+The database user needs to have full rights to modify objects in the specified schema. (eg: via `GRANT CONTROL ON SCHEMA a_sno_schema TO a_sno_user;`).
+
+### SQL Server limitations
+
+Almost all geospatial data can be converted to SQL Server format without losing any fidelity, but it does have the following limitations.
+
+#### Approximated types
+
+There is one type that Sno supports that has no SQL Server equivalent - the `interval`. This type is "approximated" as a string (or more precisely, as an `NVARCHAR`) in the SQL Server working copy while keeping its original type in the Sno dataset. Sno creates a column of type `NVARCHAR` in the appropriate place in the database table, and fills it with with the intervals formatted as [ISO8601 durations](https://en.wikipedia.org/wiki/ISO_8601#Durations). When the working copy is committed, Sno converts the contents of those columns back to `interval`, instead of their actual type, `NVARCHAR`. Since the change from `interval` to `NVARCHAR` is just a limitation of the working copy, this apparent change in type will not show up as a change in the commit log, nor will it show up as an uncommitted change if you run `sno status` to see what local changes you have made but not yet committed.
```

Comment:
Yeah some schema validation exists but is only half done - I'll add some more for intervals. It can take a while for me to get to these things since nobody in practise is checking out data with intervals into working copies that don't support it and then committing schema violating changes.
Yet. I'll write another PR.

Reply:
Within reason doesn’t really matter what the behaviour is as long as it’s documented.

As we add more patterns to our repertoire we’ll need to think less and less hopefully.

Reply:
I've expanded the schema validation to validate dates, times, timestamps and intervals, according to the V2 spec.

Sno -> SQL Server
NULL -> NULL
"P1D" -> "P1D" (1 day)

SQL Server -> Sno
NULL -> NULL
"P1D" -> "P1D"
"", "abc", etc -> In column '{col.name}' value {repr(value)} is not an ISO 8601 duration ie PxYxMxDTxHxMxS

I could add more detail here? Or I could even write a separate document that explains approximated types more fully, and link to it from all 3 working copy documents (since they all have approximated types)

Reply:
Linking to a central page with details on the various types makes sense to me.

eg:

> ### Time Interval
>
> A time interval is the intervening time between two time points, as modelled by an [ISO 8601 Duration](https://en.wikipedia.org/wiki/ISO_8601#Durations).
> 
> Natively supported in: PostgreSQL (`interval`)
> Approximated (mapped?) in: GeoPackage; MS SQL Server
>
> ... some text to describe the approximation and how it works ...